### ISSUE TITLE
Breaks with NamedModulesPlugin

### ISSUE THREAD

**Issue by cletusw** [webpack-contrib/extract-text-webpack-plugin#648]
`extract-text-webpack-plugin` breaks when combined with `webpack.NamedModulesPlugin`. It incorrectly outputs the result of extracting the CSS from the first bundle in all assets.

**main.css**
```css
body {
  background: blue;
}
```

**other.css**
```css
body {
  border: 10px solid yellow;
}
```

becomes

**main.built.css**
```css
body {
  background: blue;
}
```

**other.built.css**
```css
body {
  background: blue;
}
```

`node --version`: v8.5.0
`npm --version`: 5.5.1
`npm ls --depth=0`:
```
├── extract-text-webpack-plugin@3.0.1
├── raw-loader@0.5.1
└── webpack@3.7.1
```

EDIT: Minimal repro case: https://github.com/cletusw/extract-text-webpack-plugin-issue-648

**Comment by cletusw** [comment]
I've verified that it is not [this](https://github.com/webpack/webpack/issues/4613#issuecomment-325178346) issue. `module.libIdent` is unique for each module.

**Comment by cletusw** [comment]
Problem could be [this line](https://github.com/webpack-contrib/extract-text-webpack-plugin/blob/4cfde50a137ef83a79935467962702a0121e6d88/src/loader.js#L102), which is comparing two IDs, one of which is a number and the other is the NamedModulesPlugin-generated ID (ie, `./node_modules/raw-loader/index.js!./main.css`). Without the plugin they're both numbers so the `if` block exectues.

**Comment by cletusw** [comment]
[This line](https://github.com/webpack-contrib/extract-text-webpack-plugin/blob/4cfde50a137ef83a79935467962702a0121e6d88/src/loader.js#L98), added in 7fc7957394e2c1744737b537c2ac1bd3ba1a8046, assumes the first module will have ID `0` instead of a string. Not sure how to fix that.

**Comment by sgal** [comment]
Seems like it's a common issue to any ID-generating plugins. https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/642 could be related.

**Comment by sgal** [comment]
Looks like https://github.com/webpack-contrib/extract-text-webpack-plugin/pull/601 fixes this issue. But for some reason it was abandoned due to lack of review from @sokra

**Comment by sokra** [comment]
Sorry

**Comment by cletusw** [comment]
Fixed by d5a1de21a65532f66b05a1f6c2146574ce385567
